FlowCrypt/Controllers/MessageList Extension/MsgListViewController.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  MsgListViewController.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 23/12/2019.
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import UIKit
+
+protocol MsgListViewController {
+    func open(with message: InboxRenderable, path: String)
+
+    func getUpdatedIndex(for message: InboxRenderable) -> Int?
+    func updateMessage(isRead: Bool, at index: Int)
+    func removeMessage(at index: Int)
+}
+
+extension MsgListViewController where Self: UIViewController {
+    func open(with message: InboxRenderable, path: String) {
+        switch message.wrappedType {
+        case .message(let message):
+            openMsgElseShowToast(with: message, path: path)
+        case .thread(let thread):
+            openThread(with: thread)
+        }
+    }
+
+    // TODO: uncomment in "sent message from draft" feature
+    private func openDraft(with message: Message) {
+        guard let email = DataService.shared.email else { return }
+
+        Task {
+            let controller = await ComposeViewController(email: email)
+            await controller.updateWithMessage(message: message)
+            await navigationController?.pushViewController(controller, animated: true)
+        }
+    }
+
+    private func openMsgElseShowToast(with message: Message, path: String) {
+        if message.size ?? 0 > GeneralConstants.Global.messageSizeLimit {
```

Comment:
@tomholub 
We use `GeneralConstants.Global.messageSizeLimit` here

Reply:
Not supposed to be following https://github.com/FlowCrypt/flowcrypt-ios/commit/03e6f3551bd15dc990e97c0c55ea5aa9d670d794

This branch seems to ignore that change

Reply:
Also used in `openThread(with thread: MessageThread)` and `ThreadDetailsViewController` does not support progress loading. So we have to keep that constant. I will fix `MsgListViewController` merge.

Reply:
I guess you could merge it without progress loading for threads, and then update `ThreadDetailsViewController` to also support progress loading in another PR.